### PR TITLE
netcdf-fortran: Adapt pkg for openEuler

### PR DESCRIPTION
**Pull request by Yikun**
Signed-off-by: Yikun Jiang <yikunkero@gmail.com>

### CODE REVIEW

**Comment by Yikun** [comment]
```
No matching package to install: 'netcdf-gnu12-openmpi4-ohpc'
No matching package to install: 'phdf5-gnu12-openmpi4-ohpc >= 1.8.8'
```

File: tests/ci/spec_to_test_mapping.py [hunk 1]
```diff
@@ -198,6 +198,11 @@
         '',
         ''
     ],
+    'components/io-libs/netcdf-fortran/SPECS/netcdf-fortran.spec': [
+        'netcdf-fortran',
```
Comment (adrianreber) [inline tests/ci/spec_to_test_mapping.py]: `netcdf-fortran` does not exist in the test suite. Just use `netcdf`.

**Comment by Yikun** [comment]
```
# [prun] Master compute host = cirrus-ci-task-6047974472548352
# [prun] Resource manager = slurm
# [prun] Launch cmd = mpirun ./C_parallel atest (family=openmpi4)
# [cirrus-ci-task-6047974472548352:36254:0:36254] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x3fdef3e0)
# ==== backtrace (tid:  36254) ====
#  0  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(ucs_handle_error+0x250) [0xffffa8064720]
#  1  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(+0x25880) [0xffffa8064880]
#  2  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(+0x25c10) [0xffffa8064c10]
#  3  linux-vdso.so.1(__kernel_rt_sigreturn+0) [0xffffac9a45c0]
#  4  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(mca_btl_vader_poll_handle_frag+0x144) [0xffffa880bb88]
#  5  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(+0x4dfc) [0xffffa880bdfc]
#  6  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(opal_progress+0x3c) [0xffffab9415a0]
#  7  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_nextcid+0x84) [0xffffac167b64]
#  8  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_dup_with_info+0xf8) [0xffffac161488]
#  9  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(MPI_Comm_dup+0x6c) [0xffffac199ecc]
# 10  /opt/ohpc/pub/libs/gnu12/openmpi4/hdf5/1.10.8/lib/libhdf5.so.103(H5FD_mpi_comm_info_dup+0x68) [0xffffac5cf41c]
# 11  /opt/ohpc/pub/libs/gnu12/openmpi4/hdf5/1.10.8/lib/libhdf5.so.103(+0x353138) [0xffffac5d1138]
# 12  /opt/ohpc/pub/libs/gnu12/openmpi4/hdf5/1.10.8/lib/libhdf5.so.103(+0x21eee8) [0xffffac49cee8]
# 13  /opt/ohpc/pub/libs/gnu12/openmpi4/hdf5/1.10.8/lib/libhdf5.so.103(H5P_set+0xe8) [0xffffac4b2e88]
# 14  /opt/ohpc/pub/libs/gnu12/openmpi4/hdf5/1.10.8/lib/libhdf5.so.103(H5P_set_driver+0xf0) [0xffffac49dd90]
# 15  /opt/ohpc/pub/libs/gnu12/openmpi4/hdf5/1.10.8/lib/libhdf5.so.103(H5Pset_fapl_mpio+0xa8) [0xffffac5d1d48]
# 16  /opt/ohpc/pub/libs/gnu12/openmpi4/netcdf/4.9.0/lib/libnetcdf.so.19(NC4_create+0x178) [0xffffac730bf8]
# 17  /opt/ohpc/pub/libs/gnu12/openmpi4/netcdf/4.9.0/lib/libnetcdf.so.19(NC_create+0x264) [0xffffac6e2f54]
# 18  /opt/ohpc/pub/libs/gnu12/openmpi4/netcdf/4.9.0/lib/libnetcdf.so.19(nc_create_par+0x58) [0xffffac6fcaf8]
# 19  ./C_parallel() [0x400eec]
# 20  /usr/lib64/libc.so.6(+0x2affc) [0xffffabfa7ffc]
# 21  /usr/lib64/libc.so.6(__libc_start_main+0x94) [0xffffabfa80c8]
# 22  ./C_parallel() [0x400cb0]
```

Part of tests failed seems related to CI container without RDMA, then ucx is not working.

@adrianreber Do you have any idea to resolved this problem?

**Comment by adrianreber** [comment]
So it seems the OpenMPI BTL vader does not work in the container. This is not related to RDMA.

Switching to another BTL makes the test pass. I am not sure how this could be done in a transparent way.

```
export OMPI_MCA_btl="self,tcp"
sudo ... --preserve-env=OMPI_MCA_btl
```
makes it work for me. So instead of shared memory for local communication it is using TCP. For testing that is acceptable. The main question is why does it not work in the container even if running with `--privileged`.

Not sure if we should just always set it or not.

**Comment by adrianreber** [comment]
I was able to get it to work just by updating phdf5 to 1.12.2.

**Comment by Yikun** [comment]
> I was able to get it to work just by updating phdf5 to 1.12.2.

test passed with `docker --privileged` but without OMPI_MCA_btl?

**Comment by adrianreber** [comment]
Yes, netcdf needs a rebuild. Please open a PR to update phdf5 and probably hdf5 to a new version. Then the netcdf tests should work.

At this point I think we need a 3.x branch in GitHub because we are getting more SO name upgrades which we probably do not want to introduce in the 2.7 release.

**Comment by adrianreber** [comment]
Can you rebase on 3.x?

**Comment by adrianreber** [comment]
I have fix almost ready for the LEAP error.

**Comment by Yikun** [comment]
> I have fix almost ready for the LEAP error.

@adrianreber Thanks for your help! Feel free to update this patch if needed (I `Allow edits and access to secrets by maintainers`)

Looks like we still need `OMPI_MCA_btl` to pass test in GA, so I added it in: https://github.com/openhpc/ohpc/pull/1681/commits/18bd50f65d522040221c1eaae493cbf9f34efcd0 (Thanks @martin-g help)

**Comment by adrianreber** [comment]
Please rebase and the LEAP error should be gone.

File: tests/libs/netcdf/tests-parallel/test_pnetcdf [hunk 2]
```diff
@@ -14,6 +14,9 @@ rm=$RESOURCE_MANAGER
 testname="libs/NetCDF"
 delimiter="$RESOURCE_MANAGER/$LMOD_FAMILY_COMPILER/$LMOD_FAMILY_MPI"
 
+if [ ! -z "$SIMPLE_CI" ]; then
+    export OMPI_MCA_btl="self,tcp"
+fi
```
Comment (adrianreber) [inline tests/libs/netcdf/tests-parallel/test_pnetcdf:19]: Sorry, I have just seen this now. Is this still necessary? Even with the latest version of packages.

Comment (martin-g) [reply tests/libs/netcdf/tests-parallel/test_pnetcdf:19]: Let's try without it!

Comment (Yikun) [reply tests/libs/netcdf/tests-parallel/test_pnetcdf:19]: Yes, it was still failed before, let's see results.

Comment (martin-g) [reply tests/libs/netcdf/tests-parallel/test_pnetcdf:19]: It is still needed - https://github.com/openhpc/ohpc/actions/runs/4445282348/jobs/7804302549?pr=1681

Comment (adrianreber) [reply tests/libs/netcdf/tests-parallel/test_pnetcdf:19]: Let me also test it locally. I think I saw it working without it.

Comment (adrianreber) [reply tests/libs/netcdf/tests-parallel/test_pnetcdf:19]: not sure what I did to get it work. doesn't work any more locally without that change

**Pull request merged**